### DISCUSSION TITLE
When I set "drawOptions().prepareMolsBeforeDrawing=False", I still et error "KekulizeException: Can't kekulize mol."

### DISCUSSION THREAD

**Discussion by mohammad-saber** [rdkit/rdkit#4084] [Q&A] [answered]
Hi, thank you for reading my question.

I am trying to visualize a sub-structure of a polycyclic aromatic as below:

```
from rdkit import Chem
from rdkit.Chem import Draw
from IPython.display import SVG

smile = 'c12ccccc1N(c(c)c)c(c)c1c(c)Nc(c)c(cc)B21'
mol = Chem.MolFromSmiles(smile, sanitize=False)       
Chem.SanitizeMol(mol, sanitizeOps=(Chem.SanitizeFlags.SANITIZE_ALL^Chem.SanitizeFlags.SANITIZE_KEKULIZE^Chem.SanitizeFlags.SANITIZE_SETAROMATICITY))

d2d = Draw.MolDraw2DSVG(250,200)
d2d.drawOptions().prepareMolsBeforeDrawing=False
d2d.DrawMolecule(mol)
d2d.FinishDrawing()
SVG(d2d.GetDrawingText())
```

And I get the error:

`KekulizeException: Can't kekulize mol.  Unkekulized atoms: 0 1 2 3 4 5 10 12 13 16 18`

I thought when I set "drawOptions().prepareMolsBeforeDrawing=False", it doesn't kekulize atoms before drawing. However, I still get an error.

If you have any idea how to solve this issue, I really appreciate it to guide me. Thank you.

**Comment by ptosco** [comment, marked as answer]
@mohammad-saber Looking into the C++ code of `MolDraw2D::setupDrawMolecule` it appears that `MolDraw2DUtils::prepareMolForDrawing` is called even if `drawOptions().prepareMolsBeforeDrawing` is set to `False` in case the molecule has no conformers.
If you generate 2D coordinates for the molecule before calling `DrawMolecule`, `MolDraw2DUtils::prepareMolForDrawing` will not be called and you will be able to depict the molecule with delocalized bonds:
```
rdDepictor.Compute2DCoords(mol)
d2d = Draw.MolDraw2DSVG(250,200)
d2d.drawOptions().prepareMolsBeforeDrawing=False
d2d.DrawMolecule(mol)
d2d.FinishDrawing()
SVG(d2d.GetDrawingText())
```
![image](https://user-images.githubusercontent.com/5244385/115997366-22817200-a5e3-11eb-8264-bdde3e75c224.png)

**Reply by mohammad-saber** [reply]
Thanks , it works fine.
